cdviz-collector/src/sinks/http.rs

```diff
@@ -44,19 +45,53 @@ impl HttpSink {
 impl Sink for HttpSink {
     //TODO use cloudevents
     async fn send(&self, msg: &Message) -> Result<()> {
-        let json = serde_json::to_value(&msg.cdevent)?;
-        let resp = self
-            .client
-            .post(self.dest.clone())
-            .json(&json)
-            .send()
-            .await?;
-        if !resp.status().is_success() {
-            tracing::warn!(
-                cdevent = ?msg.cdevent,
+        if msg.cd_event.is_some() {
+            // convert  CdEvent to cloudevents
+            let value = cloudevents::EventBuilderV10::new()
+                .id(msg.cd_event.clone().unwrap().id())
+                .ty(msg.cd_event.clone().unwrap().ty())
+                .source(msg.cd_event.clone().unwrap().source().as_str())
+                .subject(msg.cd_event.clone().unwrap().subject().id())
+//                .time(msg.CdEvent.clone().unwrap().timestamp())
+                .data("application/json", serde_json::to_value(msg.cd_event.clone().unwrap())?)
+                .build();
```

Comment:
cdevents sdk-rust already provide a way to convert between cloudevents and cdevent.